aeron-client/src/main/c/util/aeron_strutil.h

```diff
@@ -58,6 +61,64 @@ inline uint64_t aeron_fnv_64a_buf(uint8_t *buf, size_t len)
  */
 int aeron_tokenise(char *input, const char delimiter, const int max_tokens, char **tokens);
 
+/**
+ * Compare two strings for equality up to the specified length. The comparison stops upon first non-equal character,
+ * if the null-character is found in one of the strings or the max length reached.
+ *
+ * @param str1 first string to compare.
+ * @param str2 second string to compare.
+ * @param length number of characters to compare.
+ * @return true if strings are equal up to the specified length.
+ */
+inline bool aeron_strn_equals(const char *str1, const char *str2, const size_t length)
+{
+    if (str1 == str2)
+    {
+        return true;
+    }
+    else if (NULL == str1 || NULL == str2)
+    {
+        return false;
+    }
+
+    for (size_t i = 0; i < length; i++)
+    {
+        if (str1[i] != str2[i])
+        {
+            return false;
+        }
+        else if (str1[i] == '\0')
+        {
+           return true;
+        }
```

Comment:
Is the `@return` documentation correct for `aeron_strn_equals("foo", "foobar", 6)`?

Reply:
Updated the doc.